src/loaders/compute_tools/tool_common.py

```diff
@@ -327,63 +321,35 @@ def parallel_batch_execution(self, tool_callable: Callable[[Dict[str, GenomeTupl
-        # distribute genome ids evenly across batches
-        chunk_size = math.ceil(len(self._data_ids) / num_batches)
-        batch_input = []
-        metas = []
-        unzipped_files_to_delete = list()
-        for batch_number, i in enumerate(range(0, len(self._data_ids), chunk_size)):
-            data_ids = self._data_ids[i: i + chunk_size]
-            batch_dir, meta = _prepare_tool(
-                self._work_dir,
-                batch_number,
-                self._node_id,
-                data_ids,
-                self._source_data_dir,
-                self._source_file_ext,
-                self._allow_missing_files,
-                self._tool_data_id_from_filename,
-                self._suffix_ids,
-            )
+        batch_dir, genomes_meta, unzipped_files_to_delete = self._prepare_execution(unzip)
 
-            if unzip:
-                unzipped_files_to_delete.extend(_unzip_files(meta))
+        ids_to_files = dict()
+        for data_id, m in genomes_meta.items():
+            # use the uncompressed file if it exists, otherwise use the source file
+            source_file = m.get(loader_common_names.META_UNCOMPRESSED_FILE,
+                                m[loader_common_names.META_SOURCE_FILE])
+            ids_to_files[m[loader_common_names.META_TOOL_IDENTIFIER]] = GenomeTuple(source_file, data_id)
 
-            metas.append((meta, batch_dir))
-            ids_to_files = dict()
-            for data_id, m in meta.items():
-                # use the uncompressed file if it exists, otherwise use the source file
-                source_file = m.get(loader_common_names.META_UNCOMPRESSED_FILE,
-                                    m[loader_common_names.META_SOURCE_FILE])
-                ids_to_files[m[loader_common_names.META_TOOL_IDENTIFIER]] = GenomeTuple(source_file, data_id)
-
-            batch_input.append((ids_to_files, batch_dir, self._program_threads, self._debug))
+        batch_input = [(ids_to_files, batch_dir, self._threads_per_tool_run, self._debug)]
 
         try:
-            self._execute(num_batches, tool_callable, batch_input, start, True)
+            self._execute(tool_callable, batch_input, start, True)
         finally:
-            if unzipped_files_to_delete:
-                print(f"Deleting {len(unzipped_files_to_delete)} unzipped files: {unzipped_files_to_delete[:5]}...")
-                for file in unzipped_files_to_delete:
-                    os.remove(file)
-
-        for meta in metas:
-            create_metadata_file(*meta)
+            self._finalize_execution(unzipped_files_to_delete, genomes_meta, batch_dir)
 
     def _execute(
             self,
-            threads: int,
             tool_callable: Callable[..., None],
             args: List[Tuple[Dict[str, GenomeTuple], Path, int, bool]] | List[Tuple[str, str, Path, Path, int, bool]],
             start: datetime.datetime,
             total: bool,
+            process_count: int = 1,
```

Comment:
Is there a reason not to just hardcode this into the Pool call? It seems like we don't even need the pool anymore

Reply:
👍 hardcoded it.

Reply:
is there a simpler way of using multiprocess to run serial processes than with a Pool?

Reply:
yea. I think we don't need to use multiprocess at all. Just `tool_callable(*arg)` should work.

Reply:
:+1: